kernel/kernel-api/src/test/scala/io/delta/kernel/internal/snapshot/LogSegmentSuite.scala

```diff
@@ -40,74 +41,86 @@ class LogSegmentSuite extends AnyFunSuite with MockFileSystemClientUtils {
   }
 
   test("constructor -- valid case (non-empty)") {
```

Comment:
test the constructor validations for checksum in test cases here? seems like we do this for the other validations?